File: example/javalib/module/15-jni/src/foo/HelloWorld.java
package foo;

public class HelloWorld {
  // Declare a native method
  public native String sayHello();

  // Load the native library
  static {
    System.load(System.getenv("HELLO_WORLD_BINARY"));
  }

  public static void main(String[] args) {
    HelloWorld helloWorld = new HelloWorld();
    System.out.println(helloWorld.sayHello());
  }
}

File: example/kotlinlib/module/15-jni/src/foo/HelloWorld.kt
package foo

class HelloWorld {
    // Declare a native method
    external fun sayHello(): String

    // Load the native library
    companion object {
        init {
            System.load(System.getenv("HELLO_WORLD_BINARY"))
        }
    }
}

fun main(args: Array<String>) {
    val helloWorld = HelloWorld()
    println(helloWorld.sayHello())
}
